pg_search/src/postgres/storage/fsm.rs

```diff
@@ -958,20 +958,27 @@ pub mod v2 {
             when_recyclable: pg_sys::FullTransactionId,
             extend_with: impl Iterator<Item = pg_sys::BlockNumber>,
         ) {
+            let mut extend_with = extend_with.peekable();
+            if extend_with.peek().is_none() {
+                // caller didn't give us anything to do
+                return;
+            }
+
             // if we are creating the index, set the XID to the first normal transaction id
             // because anything garbage-collected during index creation should be immediately reusable
             let when_recyclable = if bman.is_create_index() {
+                let first_normal_xid = pg_sys::FirstNormalTransactionId.into_inner() as u64;
+                let max_xid = first_normal_xid + (MAX_SLOTS as u64 - 1);
                 pg_sys::FullTransactionId {
-                    value: pg_sys::FirstNormalTransactionId.into_inner() as u64,
+                    value: fib_hash_u64_range(
+                        *extend_with.peek().unwrap(),
+                        first_normal_xid,
+                        max_xid,
+                    ),
```

Comment:
If parallel workers recycle blocks in similar patterns (e.g., sequential ranges), they might still collide.

For example:
```
Worker 1: Recycles blocks [100, 101, 102, ...] → Hash(100) → Slot X
Worker 2: Recycles blocks [100, 103, 104, ...] → Hash(100) → Slot X (collision!)
Worker 3: Recycles blocks [200, 201, 202, ...] → Hash(200) → Slot Y
```

Would it be better to hash based on `process_id` instead?